### PR TITLE
Improve Request Queue

### PR DESCRIPTION
**Pull request by kyscott18**

### CODE REVIEW

File: packages/core/src/utils/requestQueue.test.ts [hunk 1]
```diff
@@ -119,6 +119,7 @@ test("onIdle() resolves once idle", async ({ networks, common }) => {
   queue.request({ method: "eth_chainId" });
   queue.request({ method: "eth_chainId" });
 
+  queue.clear();
```
Comment (0xOlias) [inline packages/core/src/utils/requestQueue.test.ts:122]: Why did you need to add these?

File: packages/core/src/utils/requestQueue.ts [hunk 2]
```diff
@@ -94,44 +92,31 @@ export const createRequestQueue = ({
...
 
-        network.transport
+        const p = network.transport
           .request(params)
-          .then((a) => {
-            resolve(a);
-          })
+          .then(resolve)
           .catch(reject)
           .finally(async () => {
```
Comment (0xOlias) [inline packages/core/src/utils/requestQueue.ts:99]: Does this need to be async?

**Review by 0xOlias — APPROVED** [review]

File: packages/core/src/utils/requestQueue.test.ts (hunk 1 above)
Comment (kyscott18) [reply packages/core/src/utils/requestQueue.test.ts:122]: More accurately reflects how it is used in the actual codebase

File: packages/core/src/utils/requestQueue.ts (hunk 2 above)
Comment (kyscott18) [reply packages/core/src/utils/requestQueue.ts]: Nice catch
